src/molecules/tables/filter-column.tsx

```diff
@@ -174,69 +175,32 @@ export default function FilterColumn({
-          description: 'Submit',
-          loadingContent: <div>Loading...</div>,
-          type: 'Sheet',
-          componentType: 'CustomComponent',
-          content: (
-            <>
-              selected Rows
-              <br />
-              {selectedRows.map((row) => {
-                const _row = row as Record<string, unknown>;
-                if (column.type !== 'select-async') return row;
-                const propertyName: string = column.showProperty;
-                if (row && typeof row === 'object' && propertyName in row)
-                  return (
-                    <BadgeWithDelete
-                      badgeText={_row[propertyName] as string}
-                      handleDelete={() => {
-                        const filteredRows = [
-                          ...selectedRows.filter((r) => r !== row),
-                        ];
-                        const getFilteredValues = filteredRows
-                          .map((row) => {
-                            const _row = row as Record<string, unknown>;
-                            if (column.type !== 'select-async') return row;
-                            const propertyName: string = column.filterProperty;
-                            if (
-                              row &&
-                              typeof row === 'object' &&
-                              propertyName in row
-                            )
-                              return _row[propertyName];
-                            return row;
-                          })
-                          .join(',');
-                        setSelectedRows(filteredRows);
-                        setFilteredValue(getFilteredValues);
-                      }}
-                    />
-                  );
-                return 'not found';
-              })}
-              <DataTable
-                columnsData={{
-                  type: 'Auto',
-                  data: {
-                    ...autoColumnData,
-                    selectable: true,
-                    onSelect({ row, value, all }) {
-                      if (value === false && all) {
-                        setSelectedRows([]);
-                        setFilteredValue('');
-                        return;
-                      }
-                      if (value === false) {
-                        setSelectedRows([
-                          ...selectedRows.filter((r) => r !== row),
-                        ]);
-                        setFilteredValue(
-                          [...selectedRows.filter((r) => r !== row)]
+      {column.type === 'select-async' ? (
+        <CustomTableActionDialog
+          type="Sheet"
+          action={{
+            cta: 'Submit',
+            description: 'Submit',
+            loadingContent: <div>Loading...</div>,
+            type: 'Sheet',
+            componentType: 'CustomComponent',
+            content: (
+              <>
+                selected Rows
```

Comment:
- [ ] remove hardcoded text add resource option.

Reply:
I am planning to check the hard coded text for the whole at the end, when we see this feature works in our product.